src/log_addr.rs

```diff
@@ -24,18 +24,22 @@ pub(crate) fn pre_process<T>(
     remote_addr: Option<SocketAddr>,
 ) {
     let remote_addrs = if opts.log_remote_address {
+        let real_remote_addr =
```

Comment:
This looks a bit convoluted to me. I considered creating a function something like this?
```rust
fn get_remote_addr(remote: Option<SocketAddrs>, trusted_proxies: Vec<IpAddr>) -> String
```